test/unit/spec/nbextensions/codeCell/main-spec.js

```diff
@@ -0,0 +1,158 @@
+                type: 'code',
+                index: 1,
+                cell: newCell,
+                data: {
+                    type: 'code'
+                }
+            });
+            // there's no other triggers except to wait a moment
+            // for the cell to get turned into a bulk import cell
+            // and if it takes more than 100ms, it SHOULD fail.
+            setTimeout(() => {
+                expect(isCodeCell(newCell)).toBeTruthy();
+                done();
+            }, 100);
+        });
+
+        it('should not turn a plain code cell into a kbase code cell', (done) => {
+            const newCell = Mocks.buildMockCell('code');
+            Jupyter.notebook.cells.push(newCell);
+            Main.load_ipython_extension();
+            $([Jupyter.events]).trigger('insertedAtIndex.Cell', {
+                type: 'code',
+                index: 1,
+                cell: newCell,
+                data: {}
+            });
+            // there's no other triggers except to wait a moment
+            // for the cell to get turned into a bulk import cell
+            // and if it takes more than 100ms, it SHOULD fail.
+            setTimeout(() => {
+                expect(newCell.cellType).toBeUndefined();
+                expect(isCodeCell(newCell)).toBeFalsy();
+                done();
+            }, 100);
+        });
+
+        it('should convert userSettings into user-settings', (done) => {
+            const newCell = Mocks.buildMockCell('code', 'codeWithUserSettings');
+            expect(newCell.metadata.kbase.codeCell.userSettings.showCodeInputArea).toBe(true);
+            Jupyter.notebook.cells.push(newCell);
+
+            Main.load_ipython_extension();
+            $([Jupyter.events]).trigger('insertedAtIndex.Cell', {
+                type: 'code',
+                index: 1,
+                cell: newCell,
+                data: {
+                    type: 'code'
+                }
+            });
+            // there's no other triggers except to wait a moment
+            // for the cell to get turned into a bulk import cell
+            // and if it takes more than 100ms, it SHOULD fail.
+            setTimeout(() => {
+                expect(isCodeCell(newCell)).toBeTruthy();
+                expect(newCell.metadata.kbase.codeCell.userSettings).not.toBeDefined();
+                expect(newCell.metadata.kbase.codeCell).toEqual({
+                    'user-settings': {
+                        showCodeInputArea: true
+                    }
+                });
+                done();
+            }, 100);
+        });
+
+        it('when in doubt, use the old settings', (done) => {
+            const newCell = Mocks.buildMockCell('code', 'codeWithUserSettings');
+            newCell.metadata.kbase.codeCell['user-settings'] = { showCodeInputArea: false };
+            expect(newCell.metadata.kbase.codeCell.userSettings.showCodeInputArea).toBe(true);
```

Comment:
In case we get into a situation where a cell has both `userSettings` and `user-settings`, this ensures that the older version, `userSettings`, gets used.